src/Illuminate/Support/Url.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Stringable;
+
+class Url implements Arrayable, Stringable
+{
+    /**
+     * Constructor.
+     */
+    public function __construct(
+        public ?string $scheme = null,
+        public ?string $host = null,
+        public ?int $port = null,
+        public ?string $user = null,
+        public ?string $pass = null,
```

Comment:
Should we mark this as sensitive so it does not end up in logs?

```suggestion
        #[\SensitiveParameter]
        public ?string $pass = null,
```

Reply:
Yup I agree!